zstd_stream.go

```diff
@@ -223,24 +216,18 @@ func (w *Writer) Flush() error {
 
 	ret := 1 // So we loop at least once
 	for ret > 0 {
-		var srcPtr *byte // Do not point anywhere, if src is empty
-		if len(w.srcBuffer) > 0 {
-			srcPtr = &w.srcBuffer[0]
-		}
-
 		C.ZSTD_compressStream2_flush(
 			w.resultBuffer,
 			w.ctx,
 			unsafe.Pointer(&w.dstBuffer[0]),
 			C.size_t(len(w.dstBuffer)),
-			unsafe.Pointer(srcPtr),
-			C.size_t(len(w.srcBuffer)),
+			unsafe.Pointer(uintptr(0)),
```

Comment:
🤔 does this mean that `flush` now never does anything since we never have any data left in the source buffer?

Reply:
Flush can still do something because zstd maintains internal buffers.
i.e. if you write 1 byte in `Write()`, it's likely that zstd will buffer that internally so it can actually do compression on it.

Flush needs to be done to totally clear zstd's internal state.

Reply:
Ahhh okay, thanks for the explanation!